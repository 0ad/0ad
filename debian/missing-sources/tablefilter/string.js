/**
 * String utilities
 */

export default {

    lower(text){
        return text.toLowerCase();
    },

    upper(text){
        return text.toUpperCase();
    },

    trim(text){
        if (text.trim){
            return text.trim();
        }
        return text.replace(/^\s*|\s*$/g, '');
    },

    isEmpty(text){
        return this.trim(text) === '';
    },

    rgxEsc(text){
        let chars = /[-\/\\^$*+?.()|[\]{}]/g;
        let escMatch = '\\$&';
        return String(text).replace(chars, escMatch);
    },

    matchCase(text, mc){
        if(!mc){
            return this.lower(text);
        }
        return text;
    }

};
