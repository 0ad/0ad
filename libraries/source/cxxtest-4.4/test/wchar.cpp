//
// This program is used to check if the compiler supports basic_string<wchar_t>
//
#include <string>

int main()
{
    std::basic_string<wchar_t> s(L"s");
    return 0;
}

