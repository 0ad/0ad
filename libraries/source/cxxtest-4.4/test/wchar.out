Running cxxtest tests (1 test)
In WideCharTest::testWideStringTraits:
WideCharTest.h:14: Error: Test failed: L"std::wstring is displayed with L\"\""
WideCharTest.h:16: Error: Test failed: L"\x1234\x5678"
Failed 1 and Skipped 0 of 1 test
Success rate: 0%
Error level = 1
